docs/concept/TAP7.1-solution-design.md

```diff
@@ -0,0 +1,425 @@
+        subgraph "Enablement Provider"
+            EnablementProvider["Enablement Provider"]
+        end
+    end
+
+    EDC -.-> SSIWalletStub
+    EDC -.-> Vault
+    DataProvider --> EDC
+    DataProvider --> DTR
+    DataProvider --> SimpleSubmodelServer
+    DataConsumer --> EDC
+    EnablementProvider --> DataConsumer
+    EnablementProvider --> DataProvider
+
+````
+
+#### Deployment strategies
+
+##### Role: Data Provider
+
+The standard data provider role consists of a EDC, DTR and Submodel Server Bundles. EDC uses the Identity Wallet Bundle.
+
+````mermaid
+flowchart BT
+    A["Capability Bundle: Identity & Trust"]
+    B["Capability Bundle: Data Persistence Layer"]
+    C["Capability Bundle: Dataspace Connector"]
+    D["Capability Bundle: Digital Twin"]
+    E["Application to be tested/developed"]
+    E --> B
+    E --> C
+    E --> D
+    C --> A
+````
+
+##### Role: Data Consumer
+
+The standard data consumer role consists of only the EDC Bundle and the Identity Wallet Bundle.
+
+````mermaid
+flowchart BT
+    A["Capability Bundle: Identity & Trust"]
+    B["Capability Bundle: Dataspace Connector"]
+    C["Application to be tested/developed"]
+    C --> B
+    B --> A
+
+````
+
+##### Role: Enablement Provider
+
+The Enablement Provider roles consists of both the Data Consumer and Data Provider bundles. This allows provisioning of
+data and the consumtion of the provisioned data by a application.
+
+````mermaid
+flowchart BT
+    A["Capability Bundle: Identity & Trust"]
+    B["Capability Bundle: Data Persitance Layer"]
+    C["Application to be tested/developed"]
+    D["Capability Bundle: Digital Twin"]
+    E["Capability Bundle: Dataspace Connector (Provider)"]
+    F["Capability Bundle: Dataspace Connector (Consumer)"]
+    C --> B
+    C --> D
+    C --> E
+    C --> F
+    E --> A
+    F --> A
+````
+
+##### Bring you own
+
+Each Bundle can be deactivated to use your own component. This requires configuration of the component specific
+parameters in the other components.
+
+In this example this would be the configuration of the Identity Wallet within the EDC.
+
+````mermaid
+flowchart BT
+    A["Application to be tested/developed"]
+    B["Capability Bundle: Dataspace Connector"]
+    C["<s>Capability Bundle: Identity & Trust</s>"]
+    D["Own Identity Wallet"]
+    A --> B
+    B -- identitywallet - bundle . enabled = false --> C
+    B -- configure --> D
+
+````
+
+## 4.4 System
+
+### 4.4.1 Context Delimitation / 4.4.2 Interface Documentation
+
+````mermaid
+flowchart LR
+    Dataprovider["Dataprovider"] -->|stores data| SubmodelServer["Submodel Server"]
+    Dataprovider -->|creates twins| DTR["DTR"]
+    Dataprovider -->|creates contracts| EDC["EDC"]
+    Dataconsumer["Dataconsumer"] -->|negotiate contracts| EDC
+    DTR -->|needs| IdentityWallet["Identity Wallet"]
+    IdentityWallet -->|needs| Postgres["PostgreSQL Database"]
+````
+
+# 5 Build
+
+## 5.1 DevOps Setup Definition
+
+### 5.1.1 Tool Chain Definition
```

Comment:
Are we sure to deliver all of this? Is this mentioned in the requirements or do we extend the scope?

Reply:
Those are the tools we need to build and publish helm charts. Most of it is already there